apps/remix-ide/src/app/panels/file-panel.js

```diff
@@ -163,25 +179,52 @@ module.exports = class Filepanel extends ViewPlugin {
     })
   }
 
-  setWorkspace (workspace) {
-    const workspaceProvider = this.fileProviders.workspace
+  saveRecent(workspaceName) {
+    if (!localStorage.getItem('recentWorkspaces')) {
+      localStorage.setItem('recentWorkspaces', JSON.stringify([ workspaceName ]))
+    } else {
+      let recents = JSON.parse(localStorage.getItem('recentWorkspaces'))
+      // checking if we have a duplication
+      if (!recents.find((el) => {
+        return el === workspaceName
+      })) {
+        recents = ([workspaceName, ...recents])
+        recents.filter((el) => { return el != "" })
```

Comment:
this does nothing

Reply:
it removes workspaces with "" names

Reply:
check the documentation of the function `filter`, in your code you aren't using its return value.